### ISSUE TITLE
tokio 0.1.8 contains a breaking change

### ISSUE THREAD

**Issue by jonas-schievink** [tokio-rs/tokio#574]
Comparing the difference between 0.1.7 and 0.1.8 (https://github.com/tokio-rs/tokio/compare/tokio-0.1.7...tokio-0.1.8) I saw that the minimum supported Rust version was bumped from 1.21.0 to 1.25.0, which is considered a breaking change.

This inadvertently [broke CI for xmlrpc](https://travis-ci.org/jonas-schievink/xml-rpc-rs/jobs/420211732), which still supports Rust 1.21.0.

**Comment by carllerche** [comment]
tl;dr Tokio does not consider this a breaking change.

The change was explicitly made [here](https://github.com/tokio-rs/tokio/pull/465).

Tokio follows the same policy as rust-lang libs, which is to support a Rust version at least 3 months old (i.e. 2 release cycles ago).

It is true that this is not an ideal situation, but IMO it is the best that can be done. It has been [hotly debated](https://github.com/tokio-rs/tokio/pull/465).

That said, Rust is addressing the problem w/ an introduction of an [LTS](https://github.com/rust-lang/rfcs/pull/2483) channel. At this point, Tokio will switch to officially support the latest LTS.

Hope this clarifies things and sorry for the inconvenience.

**Comment by jonas-schievink** [comment]
Oh, okay. Everyone I've ever talked to said that requiring a newer Rust version is a breaking change.

**Comment by carllerche** [comment]
Do you have any official references to point to? My understanding is the
release train strategy implies the opposite: it is expected to float on the
latest stable.

The LTS channel won’t change this fact, it will just codify the policy to 6
months (instead of 3 that Tokio and others follow)

**Comment by jonas-schievink** [comment]
No, I've been asking on IRC (multiple times) because I wanted to decide on a Rust versioning policy for my own crates. Additionally, I've observed a handful of popular crates supporting *really* old Rust versions that also consider bumping the required version a breaking change. I haven't found much information on the "official" policy for rust-lang(-nursery) crates either.

That said, supporting the latest 3 versions does make sense and should not cause any problems for users, so I've now adopted this policy.
